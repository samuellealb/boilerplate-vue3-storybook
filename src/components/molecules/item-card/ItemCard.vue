<template>
  <div class="item-card">
    <div class="item-card__title">{{item.attributes.Name}}</div>
    <div class="item-card__button">
      <simple-button size="small" title="Delete" @click="onDelete(item.id)" />
    </div>
  </div>
</template>

<script>
import SimpleButton from '@/components/atoms/simple-button/SimpleButton.vue'
import './item-card.css'

export default {
  name: 'ItemCard',
  components: {
    SimpleButton
  },
  props: {
    item: {
      type: Object,
      required: true,
      default: () => ({id: '', attributes: {Name: ''}}),
      validator: item => ['id', 'attributes'].every(key => key in item),
    }
  },
  emits: ['delete-item'],
  setup(props, { emit }) {
    function onDelete (id) {
      emit('delete-item', id);
    }
    return { onDelete }
  },
}
</script>