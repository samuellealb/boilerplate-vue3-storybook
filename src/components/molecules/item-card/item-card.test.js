import { mount } from '@vue/test-utils'
import { test, expect } from 'vitest'
import ItemCard from './ItemCard.vue'

test('mount component', async () => {
  expect(ItemCard).toBeTruthy()

  const defaultItemCard = mount(ItemCard, {
    props: {
      item: {
        id: '2',
        attributes: {
          Name: 'Testing Item Card'
        }
      }    
    }
  })
  expect(defaultItemCard.html()).toMatchSnapshot()

})

test('Click on button emit "delete-item" with item id value', async() => {
  const wrapper = mount(ItemCard, {
    props: {
      item: {
        id: 1,
        attributes: {
          Name: 'Testing Item Card'
        }
      }    
    }
  })

  wrapper.find('button').trigger('click')
  await wrapper.vm.$nextTick()
  expect(wrapper.emitted('delete-item')[0]).toEqual([1])
})