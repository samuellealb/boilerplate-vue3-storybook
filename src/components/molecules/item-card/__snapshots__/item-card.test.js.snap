// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"item-card\\">
  <div class=\\"item-card__title\\">Testing Item Card</div>
  <div class=\\"item-card__button\\"><button class=\\"simple-button simple-button--default simple-button--small\\">Delete</button></div>
</div>"
`;
